java-checks/src/main/java/org/sonar/java/checks/CognitiveComplexityMethodCheck.java

```diff
@@ -77,199 +53,11 @@
   @Override
   public void visitNode(Tree tree) {
     MethodTree method = (MethodTree) tree;
-    if (method.block() == null || ((ClassTree)method.parent()).simpleName() == null || isWithinLocalClass(method)) {
-      return;
-    }
-    CognitiveComplexityVisitor complexityVisitor = new CognitiveComplexityVisitor();
-    method.accept(complexityVisitor);
-    int total = complexityVisitor.complexity;
+    JavaFileScannerContext.CognitiveComplexity result = context.cognitiveComplexity(method);
```

Comment:
`CognitiveComplexityVisitor.methodComplexity` can be used directly, so method on `JavaScannerContext` can be removed